blockprod/src/detail/mod.rs

```diff
@@ -526,7 +592,7 @@ mod tests {
         );
 
         assert!(
-            other_job_cancel_receiver.try_recv().unwrap().is_none(),
+            other_job_cancel_receiver.try_recv().is_err(),
```

Comment:
I think is_err is not equivalent to unwrap().is_none(). The TryRecvError can be either Closed or Empty, and it makes a difference which is the current state.

Reply:
Ah of course. Ok, will update